s4e-web/cypress/integration/map/map-layers.spec.ts

```diff
@@ -1,141 +1,291 @@
+        'VERSION=x.y.z',
+        'HEIGHT=none',
+        'WIDTH=none',
+        'CRS=unknown',
+        'BBOX=unknown,unknown,unknown,unknown'
+      ].join('&');
+
+      MapLayers
+        .openManagementModal()
+        .fillFormIncorrectly(label,url)
+        .errorsCountShouldBe(6);
+      GeneralModal
+        .closeModal();
+    });
+
+    it('handle 400 error', function () {
+      const label = 'Test';
+      const geoserverUrl = '/test-wms';
+
+      cy.server();
+      cy.route({
+        method: 'GET',
+        url: `${geoserverUrl}?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetCapabilities`,
+        headers: {
+          'Content-type': 'text/xml'
+        },
+        status: 400,
+        response:  { errors: 'CannotGET /...' }
+      })
+        .as('getCapabilities');
+      MapLayers
+        .openManagementModal()
+        .fillFormCorrectly(label, geoserverUrl)
+        .errorShouldBeDispalyed();
+        GeneralModal
+        .closeModal();
+    });
+
+    it('handle 404 error', function () {
+      const label = 'Test';
+      const geoserverUrl = '/test-wms';
+
+      cy.server();
+      cy.route({
+        method: 'GET',
+        url: `${geoserverUrl}?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetCapabilities`,
+        headers: {
+          'Content-type': 'text/xml'
+        },
+        status: 404,
+        response:  { errors: 'CannotGET /...' }
+      })
+        .as('getCapabilities');
+
+      MapLayers
+        .openManagementModal()
+        .fillFormCorrectly(label, geoserverUrl)
+        .errorShouldBeDispalyed();
+        GeneralModal
+        .closeModal();
+    });
+
+    it('handle 500 error', function () {
+      const label = 'Test';
+      const geoserverUrl = '/test-wms';
+
+      cy.server();
+      cy.route({
+        method: 'GET',
+        url: `${geoserverUrl}?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetCapabilities`,
+        headers: {
+          'Content-type': 'text/xml'
+        },
+        status: 500,
+        response:  { errors: 'CannotGET /...' }
+      })
+        .as('getCapabilities');
+      MapLayers
+        .openManagementModal()
+        .fillFormCorrectly(label, geoserverUrl)
+        .errorShouldBeDispalyed();
+        GeneralModal
+        .closeModal();
+    });
+
+    it('handle 503 error', function () {
+      const label = 'Test';
+      const geoserverUrl = '/test-wms';
+
+      cy.server();
+      cy.route({
+        method: 'GET',
+        url: `${geoserverUrl}?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetCapabilities`,
+        headers: {
+          'Content-type': 'text/xml'
+        },
+        status: 503,
+        response:  { errors: 'CannotGET /...' }
+      })
+        .as('getCapabilities');
+        
+      MapLayers
+        .openManagementModal()
+        .fillFormCorrectly(label, geoserverUrl)
+        .errorShouldBeDispalyed();
+        GeneralModal
+        .closeModal();
+    });
+
+    it.only('handle response does not have the correct formats ', function () {
```

Comment:
I see 400, 404. 500, 503 and incorrect format errors handling, but CORS is still missing.